app/lib/world/thang.coffee

```diff
@@ -28,7 +28,7 @@ module.exports = class Thang
   constructor: (@world, @spriteName, @id) ->
     @spriteName ?= @constructor.className
     @id ?= @constructor.nextID @spriteName, @world
-    @addTrackedProperties ['exists', 'boolean']  # TODO: move into Systems/Components, too?
+    @addTrackedProperties ['exists', 'boolean'], ['variableNames', 'string']  # TODO: move into Systems/Components, too?
```

Comment:
Can we add this more specifically, like in the Programming System at the first moment we track a variable for a Thang? It will be a performance hit for too many Things otherwise. Other than that, this looks good.